src/Files.Uwp/Filesystem/StorageFileHelpers/StorageFileWithPath.cs

```diff
@@ -1,37 +1,22 @@
 ﻿using Files.Filesystem.StorageItems;
 using Windows.Storage;
+using IO = System.IO;
 
 namespace Files.Filesystem
 {
     public class StorageFileWithPath : IStorageItemWithPath
     {
-        public BaseStorageFile File
-        {
-            get
-            {
-                return (BaseStorageFile)Item;
-            }
-            set
-            {
-                Item = value;
-            }
-        }
+        public string Path { get; }
+        public string Name => Item?.Name ?? IO.Path.GetFileName(Path);
 
-        public string Path { get; set; }
-        public string Name => Item?.Name ?? System.IO.Path.GetFileName(Path);
-        public IStorageItem Item { get; set; }
-        public FilesystemItemType ItemType => FilesystemItemType.File;
+        IStorageItem IStorageItemWithPath.Item => Item;
+        public BaseStorageFile Item { get; }
 
-        public StorageFileWithPath(BaseStorageFile file)
-        {
-            File = file;
-            Path = File.Path;
-        }
+        public FilesystemItemType ItemType => FilesystemItemType.Directory;
```

Comment:
FilesystemItemType.File?

Reply:
It is indeed an error. Thanks I fixed it.